### PR TITLE
Add CrateDB testcontainers

### PR DESCRIPTION
**Pull request by matriv**
Addresses: #6789

### CODE REVIEW

File: modules/jdbc/src/main/java/org/testcontainers/jdbc/ConnectionUrl.java [hunk 1]
```diff
@@ -205,7 +205,7 @@ public Map<String, String> getTmpfsOptions() {
     public interface Patterns {
         Pattern URL_MATCHING_PATTERN = Pattern.compile(
             "jdbc:tc:" +
-            "(?<databaseType>[a-z0-9]+)" +
+            "(?<databaseType>[a-z0-9_]+)" +
```
Comment (eddumelendez) [inline modules/jdbc/src/main/java/org/testcontainers/jdbc/ConnectionUrl.java:208]: rollback this change

File: modules/jdbc/src/test/java/org/testcontainers/jdbc/ConnectionUrlTest.java [hunk 2]
```diff
@@ -13,10 +13,10 @@ public class ConnectionUrlTest {
 
     @Test
     public void testConnectionUrl1() {
-        String urlString = "jdbc:tc:mysql:5.7.34://somehostname:3306/databasename?a=b&c=d";
+        String urlString = "jdbc:tc:my_db:5.7.34://somehostname:3306/databasename?a=b&c=d";
```
Comment (eddumelendez) [inline modules/jdbc/src/test/java/org/testcontainers/jdbc/ConnectionUrlTest.java:16]: rollback this change

File: modules/jdbc/src/test/java/org/testcontainers/jdbc/ConnectionUrlTest.java [hunk 3]
```diff
@@ -13,10 +13,10 @@ public class ConnectionUrlTest {
...
         ConnectionUrl url = ConnectionUrl.newInstance(urlString);
 
-        assertThat(url.getDatabaseType()).as("Database Type value is as expected").isEqualTo("mysql");
+        assertThat(url.getDatabaseType()).as("Database Type value is as expected").isEqualTo("my_db");
```
Comment (eddumelendez) [inline modules/jdbc/src/test/java/org/testcontainers/jdbc/ConnectionUrlTest.java:19]: rollback this change

**Review by eddumelendez — CHANGES_REQUESTED** [review]
Thanks for your contribution, @matriv ! I have left some comments and we suggest to remove the legacy driver given it is a new module.

Also, please do not use force push.

File: modules/cratedb/src/test/java/org/testcontainers/CrateDBTestImages.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package org.testcontainers;
+
+import org.testcontainers.utility.DockerImageName;
+
+public interface CrateDBTestImages {
```
[the remark this answers is not in the document]
Comment (matriv) [reply modules/cratedb/src/test/java/org/testcontainers/CrateDBTestImages.java:5]: What do you mean: avoid doing this?
Should I use a constant in the test classes that need it instead?

**Comment by matriv** [comment]
Thx a lot for reviewing the PR @eddumelendez !
>  we suggest to remove the legacy driver given it is a new module.

There are certain places that still require the legacy driver, because if vanilla Postgres is used,
they assume things about the internal structures/tables where CrateDB is not 100% combatible, see [this example](https://github.com/apache/flink-connector-jdbc/pull/29/files#diff-2de81db8fb3556629101d31f0a22ffcf7f4e446f7f9cec24b24f56836a445d81R68)

Although, there is a workaround for such cases, so if you prefer not to include the legacy version, I'd be happy to remove it.

> Also, please do not use force push.

Apologies, I didn't change anything in my PR, just rebased with `main`.  I won't force push again of course, no that the PR is in review.

**Comment by eddumelendez** [comment]
@matriv thanks for the quick update! Please, let's remove the legacy version

**Comment by matriv** [comment]
> @matriv thanks for the quick update! Please, let's remove the legacy version

It's reasonable, thx! please also check: https://github.com/testcontainers/testcontainers-java/pull/6790#discussion_r1152070909

File: modules/cratedb/src/test/java/org/testcontainers/CrateDBTestImages.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (matriv) [reply modules/cratedb/src/test/java/org/testcontainers/CrateDBTestImages.java:5]: no worries, thx!

**Comment by matriv** [comment]
I believe I've had addressed your comments so far, removed the legacy version, please have another look, many thx!

**Review by eddumelendez — APPROVED** [review]

**Comment by eddumelendez** [comment]
Thanks for your contribution, @matriv ! This is now merged in `main` branch and it will be part of the next release.

**Comment by matriv** [comment]
Thx a lot for reviewing and merging!

**Pull request merged**